lib/streamlit/elements/multiselect.py

```diff
@@ -215,18 +216,25 @@ def _check_and_convert_to_indices(
                     default_values = [default_values]
                 else:
                     default_values = list(default_values)
-            if len(default_values) != 0 and isinstance(default_values[0], Enum):
-                str_default_values = [str(enum) for enum in default_values]
-                mapped_opt_keys = [str(enum) for enum in opt]
-                for value in str_default_values:
-                    if value not in mapped_opt_keys:
-                        raise StreamlitAPIException(
-                            "Every Multiselect default value must exist in options"
-                        )
-                return [mapped_opt_keys.index(value) for value in str_default_values]
 
             for value in default_values:
                 if value not in opt:
+                    if isinstance(value, Enum) and all(
+                        not likely_equivalent(value, option) for option in opt
+                    ):
```

Comment:
I realize it was me who suggested this conditional, but given that its wrong, its probably a good idea to fix it :sweat_smile: 

```suggestion
                    if isinstance(value, Enum) and any(
                        likely_equivalent(value, option) for option in opt
                    ):
```